sdk/keyvault/azure-security-keyvault-certificates/src/main/java/com/azure/security/keyvault/certificates/CertificateAsyncClient.java

```diff
@@ -413,45 +414,43 @@ public String getVaultUrl() {
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<DeletedCertificate> deleteCertificate(String name) {
-        try {
-            return withContext(context -> deleteCertificateWithResponse(name, context)).flatMap(FluxUtil::toMono);
-        } catch (RuntimeException ex) {
-            return monoError(logger, ex);
-        }
+    public PollerFlux<DeletedCertificate, Void> beginDeleteCertificate(String name) {
+        return new PollerFlux<>(Duration.ofSeconds(1),
+            activationOperation(name),
+            createDeletePollOperation(name),
+            (context, firstResponse) -> Mono.empty(),
+            (context) -> Mono.empty());
     }
 
-    /**
-     * Deletes a certificate from a specified key vault. All the versions of the certificate along with its associated policy
-     * get deleted. If soft-delete is enabled on the key vault then the certificate is placed in the deleted state and requires to be
-     * purged for permanent deletion else the certificate is permanently deleted. The delete operation applies to any certificate stored in
-     * Azure Key Vault but it cannot be applied to an individual version of a certificate. This operation requires the certificates/delete permission.
-     *
-     * <p><strong>Code Samples</strong></p>
-     * <p>Deletes the certificate in the Azure Key Vault. Prints out the deleted certificate details when a response has been received.</p>
-     *
-     * {@codesnippet com.azure.security.keyvault.certificates.CertificateAsyncClient.deleteCertificateWithResponse#string}
-     *
-     * @param name The name of the certificate to be deleted.
-     * @throws ResourceNotFoundException when a certificate with {@code name} doesn't exist in the key vault.
-     * @throws HttpRequestException when a certificate with {@code name} is empty string.
-     * @return A {@link Mono} containing a {@link Response} whose {@link Response#getValue() value} contains the {@link DeletedCertificate deleted certificate}.
-     */
-    @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<Response<DeletedCertificate>> deleteCertificateWithResponse(String name) {
-        try {
-            return withContext(context -> deleteCertificateWithResponse(name, context));
-        } catch (RuntimeException ex) {
-            return monoError(logger, ex);
-        }
+    private Function<PollingContext<DeletedCertificate>, Mono<DeletedCertificate>> activationOperation(String name) {
+        return (pollingContext) -> withContext(context -> deleteCertificateWithResponse(name, context)
+            .flatMap(deletedKeyResponse -> Mono.just(deletedKeyResponse.getValue())));
+    }
+
+    /*
+    Polling operation to poll on create delete key operation status.
+    */
+    private Function<PollingContext<DeletedCertificate>, Mono<PollResponse<DeletedCertificate>>> createDeletePollOperation(String keyName) {
```

Comment:
If you already have a poller for creating a certificate, do you need a separate one for canceling and deleting? For .NET, I just used the same poller, send a cancel or delete request, and then waited in the same loop for an appropriate response.

Reply:
removed cancel operation API as discussed.
delete operation API exists.